implementation/src/main/java/io/smallrye/config/lifecycle/ConfigSourceInterceptorContext.java

```diff
@@ -0,0 +1,35 @@
+package io.smallrye.config.lifecycle;
+
+public interface ConfigSourceInterceptorContext {
+    /**
+     * This will provide the ConfigSourceInterceptor with the lookup value of the key on a ConfigSource.
+     *
+     * Should the ConfigSource be exposed?
+     *
+     * @return a ConfigValue
+     */
+    ConfigValue getConfigValue();
+
+    /**
+     * Lookup a key only into the current or previously evaluated ConfigSources
+     *
+     * @param key the key to lookup.
+     * @return the lookup key value.
+     */
+    String getValue(String key);
+
+    /**
+     * Proceeds to the next interceptor.
+     *
+     * @return the config key lookup value.
+     */
+    String proceed();
```

Comment:
This would be dropped.

Reply:
This was just a shorthand to proceed if you didn't want to push a new value. In terms of API, for instance if you are writing the LoggerInterceptor, which you only want to log stuff and don't change anything, it looks weird to me that you have to .process and pass something into the method call.